contracts/governance/StakingRewards/StakingRewards.sol

```diff
@@ -146,34 +143,38 @@ contract StakingRewards is StakingRewardsStorage, Initializable {
 	 * @return The timestamp of last withdrawal
 	 * @return The accumulated reward
 	 */
-	function getStakerCurrentReward(bool considerMaxDuration) public view returns (uint256 withdrawalTime, uint256 amount) {
-		uint256 count;
+	function getStakerCurrentReward(bool considerMaxDuration) public view returns (uint256 lastWithdrawalInterval, uint256 amount) {
 		uint256 weightedStake;
-		uint256 lastFinalisedBlock = block.number - 1;
+		uint256 lastFinalisedBlock = _getCurrentBlockNumber() - 1;
 		uint256 currentTS = block.timestamp;
+		uint256 addedMaxDuration;
 		address staker = msg.sender;
-		uint256 duration;
-		if (withdrawals[staker] == 0) {
-			require((currentTS.sub(startTime)) > TWO_WEEKS, "allowed after 14 days of start");
-			withdrawalTime = startTime;
-		} else {
-			require(currentTS > withdrawals[staker], "allowed after 14 days");
-			withdrawalTime = withdrawals[staker];
-		}
+		uint256 referenceBlock;
 
-		if (considerMaxDuration) {
-			duration = maxDuration;
-		} else {
-			duration = currentTS;
-		}
+		uint256 lastStakingInterval = staking.timestampToLockDate(currentTS);
+		lastWithdrawalInterval = withdrawals[staker] > 0 ? withdrawals[staker] : startTime;
+		if (lastStakingInterval < lastWithdrawalInterval) return (0, 0);
+
+		if (considerMaxDuration) addedMaxDuration = lastWithdrawalInterval.add(maxDuration);
+		uint256 duration =
+			considerMaxDuration && (addedMaxDuration < currentTS) ? staking.timestampToLockDate(addedMaxDuration) : lastStakingInterval;
 
-		for (uint256 i = withdrawalTime; i <= currentTS && i <= withdrawalTime.add(duration); i += TWO_WEEKS) {
-			count++;
-			weightedStake = weightedStake.add(_computeRewardForDate(staker, lastFinalisedBlock, i));
+		for (uint256 i = lastWithdrawalInterval; i < duration; i += TWO_WEEKS) {
```

Comment:
is it valid if `i == duration?`

Reply:
Example: Aug 1 700 GMT - Aug 15 700 GMT
Aug 15 700 GMT - Aug 29 700 GMT
Aug 29 700 GMT - Sep 13 700 GMT

If someone collect rewards on August 30th for the first time, the lastStakingInterval will be Aug 29 700 GMT.
duration = lastStakingInterval(as per the logic)

So the loop runs twice for Aug 1 and Aug 15 and doesn't run for third time as Aug 29 700 GMT(i) is not less than Aug 29 700 GMT(duration). Hence < looks like a valid logic to me.

Reply:
but it looks like withdrawInterval is not the beginning, but the end of the interval, no?
`duration` == `lastStakingInterval`
taken withdrawal is on aug 30 at the second step in the loop, it will not count Aug 15 700 GMT - Aug 29 700 GMT as it would be equal to Aug 29

Reply:
lastWithdrawInterval is not the beginning, but the end of the interval - Yes
It actually is **timestampToLockDate**(the end of the interval) - which means, it would save the last date of staking period closest to the withdrawal date which is Aug 29 7 GMT for an August 30th withdrawal

Note that we do:
lastStakingInterval = **staking.timestampToLockDate(currentTS)**;
and duration = addedMaxDuration < currentTS ? **staking.timestampToLockDate(addedMaxDuration)** : lastStakingInterval;